### PR TITLE
🐛 Fix e2e pivoting by adding capi labels on hardware data crd

### PR DESCRIPTION
**Pull request by mboukhalfa**
Add CAPI labels on hardwareData CRD (added in https://github.com/metal3-io/baremetal-operator/pull/1099) so that hadwareData object is also pivoted as part of clusterctl move operation. Ref: https://github.com/metal3-io/metal3-dev-env/pull/969

### CODE REVIEW

**Review by fmuyassarov — COMMENTED** [review]
looks good to me. Although I noticed that for BMH we are not setting move label but rather provider name, which should be okay.
```
"clusterctl.cluster.x-k8s.io=",
"cluster.x-k8s.io/provider=metal3",
```

**Review by lentzi90 — COMMENTED** [review]
Nice find @mboukhalfa ! Thanks for fixing.

I see you are testing something with nft, are you seeing issues in the tests or what is this for?

File: scripts/ci-e2e.sh [hunk 1]
```diff
@@ -71,7 +71,7 @@ else
   export CAPM3_FROM_RELEASE="${CAPM3_FROM_RELEASE:-$(get_latest_release "${CAPM3RELEASEPATH}" "v0.5.")}"
   export CAPM3_TO_RELEASE="${CAPM3RELEASE}"
 fi
-
+sudo nft flush ruleset
```
Comment (kashifest) [inline scripts/ci-e2e.sh:74]: why do we need this?

Comment (Xenwar) [reply scripts/ci-e2e.sh:74]: could this be one of the commands to fix the broken pipe issue ?
I do not think we need it in this PR.

**Comment by mboukhalfa** [comment]
/hold cancel 
Merging this PR to fix e2e centos test
e2e-centos tested but we will not run ubuntu due to the bmo/ironic issue that will be fixed afterwards

**Comment by mboukhalfa** [comment]
> Nice find @mboukhalfa ! Thanks for fixing.
> 
> I see you are testing something with nft, are you seeing issues in the tests or what is this for?

Was trying to debug ironic bmo communication issue on ubuntu those are removed now

**Pull request merged**